### PR TITLE
Reset WorldGuard's player state when imprisoning/releasing a player

### LINKED ISSUE #39: Certain WorldGuard flags misbehave when applied to prisoner group only (via membership)
Given a region around the jail/prison area. Seems to be a caching "issue" in WorldGuard due to how BetterJails changes the prisoner's group and teleports the player in and out of the region. `/worldguard flushstates <player>` seems to fix it.

Partly BetterJails is to blame given that changing the parent groups and teleporting in/out of the region both happen asynchronously in entirely unrelated scheduling contexts (assuming Paper + LuckPerms)

### PR DESCRIPTION
**Pull request by emilyy-dev**
Tested. Fixes #39

### CODE REVIEW

File: betterjails/src/main/java/io/github/emilyydev/betterjails/data/PrisonerDataHandler.java [hunk 1]
```diff
@@ -138,16 +140,25 @@ public void addJailedPlayer(
       knownLastLocation = existingPrisoner.lastLocationNullable();
     }
 
-    if (isPlayerOnline) {
+    // this is ugly
+    final Runnable teleportAction;
```
Comment (rymiel) [inline betterjails/src/main/java/io/github/emilyydev/betterjails/data/PrisonerDataHandler.java]: making this not final removes the need for the newly added if cases

File: betterjails/src/main/java/io/github/emilyydev/betterjails/interfaces/WorldGuardFacade.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  static {
+    try {
+      final MethodHandles.Lookup lookup = MethodHandles.lookup();
+      MethodHandle resetStateMh = lookup.findStatic(WorldGuardFacade.class, "resetStateNop", MethodType.methodType(void.class, Player.class));
+      try {
+        final Class<?> sessionManagerClass = Class.forName("com.sk89q.worldguard.session.SessionManager");
+        if (sessionManagerClass.isInterface()) { // 7.x
+          resetStateMh = lookup.findStatic(WorldGuardFacade.class, "resetState0", MethodType.methodType(void.class, Player.class));
+        } // TODO: <=1.12
```
Comment (rymiel) [inline betterjails/src/main/java/io/github/emilyydev/betterjails/interfaces/WorldGuardFacade.java:48]: do you maybe wanna make an issue for this so you don't forget to do the todo

**Review by rymiel — CHANGES_REQUESTED** [review]

**Review by rymiel — APPROVED** [review]

File: betterjails/src/main/java/io/github/emilyydev/betterjails/data/PrisonerDataHandler.java (hunk 1 above)
Comment (emilyy-dev) [reply betterjails/src/main/java/io/github/emilyydev/betterjails/data/PrisonerDataHandler.java]: fixed in https://github.com/emilyy-dev/BetterJails/pull/40/commits/8631c3f112b5adc19802766c288f48b8bc3fd775

File: betterjails/src/main/java/io/github/emilyydev/betterjails/interfaces/WorldGuardFacade.java (hunk 2 above)
Comment (emilyy-dev) [reply betterjails/src/main/java/io/github/emilyydev/betterjails/interfaces/WorldGuardFacade.java:48]: maybe

**Pull request merged**